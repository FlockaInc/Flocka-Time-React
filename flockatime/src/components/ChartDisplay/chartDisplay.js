import React from 'react';
import './chartDisplay.css';

import BarGraph from '../BarGraph/barGraph';

function ChartDisplay() {

  function generateChart() {
    // getData and generate chart with d3
    return null;
  }

  return (
    <div className='row'>
      <div className='col-1'></div>
      <div className='col-10'>
        {generateChart()}
      </div>
      <div className='col-1'></div>
    </div>
  );
}

export default ChartDisplay;