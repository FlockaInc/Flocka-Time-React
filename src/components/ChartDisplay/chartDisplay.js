import React, { useState, useEffect } from 'react';
import './chartDisplay.css';

import BarGraph from '../BarGraph/barGraph';

import ns from '../../utilities/notificationService';
import ds from '../../utilities/data';

function ChartDisplay() {
  const [authenticated, setAuthenticated] = useState(false);
  const [userDailyTime, setUserDailyTime] = useState(null);

  useEffect(() => {
    ns.addObserver('AUTH_SIGNIN', this, handleSignIn);
    ns.addObserver('AUTH_SIGNOUT', this, handleSignOut);
    ns.addObserver('DATA_FLOCKALOGS_DOWNLOADED', this, handleFlockaLogData);

    return () => {
      ns.removeObserver(this, 'AUTH_SIGNIN');
      ns.removeObserver(this, 'AUTH_SIGNOUT');
      ns.removeObserver(this, 'DATA_FLOCKALOGS_DOWNLOADED');
    }
  }, []);

  function handleSignIn() {
    setAuthenticated(true);
  }

  function handleSignOut() {
    setAuthenticated(false);
  }

  function handleFlockaLogData() {
    setUserDailyTime(ds.getCurrentUserDailyFlockatime());
  }

  function generateChart() {
    if (authenticated && userDailyTime) {
      return (
        <BarGraph userDailyTime={userDailyTime} />
      )
    }

    return null;
  }

  return (
    <div className='row'>
      <div className='col-1'></div>
      <div className='col-10'>
        {generateChart()}
      </div>
      <div className='col-1'></div>
    </div>
  );
}

export default ChartDisplay;