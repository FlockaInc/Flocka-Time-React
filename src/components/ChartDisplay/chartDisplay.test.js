import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ChartDisplay from './chartDisplay';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<ChartDisplay />);
  expect(asFragment()).toMatchSnapshot();
});