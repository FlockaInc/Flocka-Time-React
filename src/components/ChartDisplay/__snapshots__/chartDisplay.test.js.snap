// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-1"
    />
    <div
      class="col-10"
    />
    <div
      class="col-1"
    />
  </div>
</DocumentFragment>
`;
